File: TI 1.1/WS_CONUNI_SOAPJAVA_G04/02 CLIESC/CLIESC_CONVUNI_SOAP_GR04/src/main/java/ec/edu/monster/model/ConversionType.java
package ec.edu.monster.model;

public enum ConversionType {
    // Longitud
    CM_A_IN("Centímetros → Pulgadas", ConversionCategory.LONGITUD),
    IN_A_CM("Pulgadas → Centímetros", ConversionCategory.LONGITUD),

    // Temperatura
    C_A_F("Celsius (°C) → Fahrenheit (°F)", ConversionCategory.TEMPERATURA),
    F_A_C("Fahrenheit (°F) → Celsius (°C)", ConversionCategory.TEMPERATURA),

    // Masa
    KG_A_LB("Kilogramos → Libras", ConversionCategory.MASA),
    LB_A_KG("Libras → Kilogramos", ConversionCategory.MASA);

    private final String label;
    private final ConversionCategory category;

    ConversionType(String label, ConversionCategory category) {
        this.label = label; this.category = category;
    }
    public String label() { return label; }
    public ConversionCategory category() { return category; }
    @Override public String toString() { return label; }
}

File: TI1.2/SOAP_DOTNET_CONUNI_GR04/02.CLIESC/CLIESC_CONVUNI_SOAP_GR04/ec.edu.monster.modelo/ConversionType.cs
using System.ComponentModel;

namespace ec.edu.monster.modelo
{
    public enum ConversionType
    {
        // Longitud
        [Description("Centímetros → Pulgadas")] CM_A_IN,
        [Description("Pulgadas → Centímetros")] IN_A_CM,

        // Temperatura
        [Description("Celsius (°C) → Fahrenheit (°F)")] C_A_F,
        [Description("Fahrenheit (°F) → Celsius (°C)")] F_A_C,

        // Masa
        [Description("Kilogramos → Libras")] KG_A_LB,
        [Description("Libras → Kilogramos")] LB_A_KG
    }
}
